### PR TITLE
feat(merchant): Added merchant schedule component

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
The schedule component renders the opening hours per week as well as any of the deviated dates. The dates are filtered based on an week offset, so you can for example show the dates from last week and the coming 12 weeks. 

closes: [HRZ-90755](https://spryker.atlassian.net/browse/HRZ-90755)

[HRZ-90755]: https://spryker.atlassian.net/browse/HRZ-90755?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dunqan — DISMISSED** [review]

File: libs/domain/merchant/schedule/schedule.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+// const lastWeek = new Date(new Date().setDate(new Date().getDate() - 6));
+// const lastMonth = new Date(new Date().setDate(new Date().getDate() - 20));
+// const lastQuarter = new Date(new Date().setDate(new Date().getDate() - 40));
+// const lastYear = new Date(new Date().setDate(new Date().getDate() - 190));
+// const twoYearsBack = new Date(new Date().setDate(new Date().getDate() - 400));
+// const aWeekAhead = new Date(new Date().setDate(new Date().getDate() + 4));
+// const aMonthAhead = new Date(new Date().setDate(new Date().getDate() + 20));
+// const aQuarterAhead = new Date(new Date().setDate(new Date().getDate() + 40));
+// const aYearAhead = new Date(new Date().setDate(new Date().getDate() + 180));
+// const twoYearsAhead = new Date(new Date().setDate(new Date().getDate() + 400));
```
Comment (dunqan) [inline libs/domain/merchant/schedule/schedule.component.spec.ts]: 👀

File: libs/domain/merchant/schedule/schedule.component.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  weeksAfter: 12,
+})
+export class MerchantScheduleComponent extends MerchantMixin(
+  ContentMixin<MerchantScheduleComponentOptions>(LitElement)
+) {
+  static styles = merchantScheduleStyles;
+
+  protected $dates = computed(() => {
+    const { weeksBefore, weeksAfter } = this.$options();
+    if (weeksBefore === undefined && weeksAfter === undefined)
```
Comment (dunqan) [inline libs/domain/merchant/schedule/schedule.component.ts:28]: will we get undefined ever if we have default options for both?

File: libs/domain/merchant/schedule/schedule.def.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { componentDef } from '@spryker-oryx/utilities';
+
+export const merchantOpeningHoursComponent = componentDef({
```
Comment (dunqan) [inline libs/domain/merchant/schedule/schedule.def.ts]:
```suggestion
export const merchantScheduleComponent = componentDef({
```

File: libs/domain/merchant/src/mixins/merchant.mixin.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    protected $merchant = computed(() => {
+      const qualifier = this.merchant
+        ? { id: this.merchant as string }
+        : this.$merchantContext();
+      return qualifier
+        ? this[MerchantMixinInternals].merchantService?.get(qualifier)
+        : undefined;
+    });
+
+    // protected $merchantMinimal = computed(() => {
```
Comment (dunqan) [inline libs/domain/merchant/src/mixins/merchant.mixin.ts]: let's clean this up, we if don't need this let's remove it also from mixin interface

File: libs/domain/site/time/time.component.ts [hunk 5]
```diff
@@ -18,19 +14,21 @@ export class SiteTimeComponent
 {
   protected localeService = resolve(LocaleService);
 
-  @signalProperty() stamp?: string | number | Date;
-  @signalProperty() i18nToken?: string;
+  @property() stamp?: string | number | Date;
```
Comment (dunqan) [inline libs/domain/site/time/time.component.ts:17]: why not signal property?

File: libs/domain/merchant/schedule/schedule.component.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply libs/domain/merchant/schedule/schedule.component.ts:28]: yeah, if you don't want to show any dates, you can set them to undefined

**Review by dunqan — DISMISSED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**